### PR TITLE
update go module dependencies

### PR DESCRIPTION
**Pull request by sk1tter**
## Update go module dependencies

Updated by running 
```bash
go get -u -d ./...
go mod tidy
```
in every directory containing `go.mod` file.

Some not updated directories are below.
- `./benchmarks/video-analytics`
- `./benchmarks/chained-function-serving`
- `./benchmarks/corral`
with error 
```
go.opentelemetry.io/otel/semconv: cannot find module providing package go.opentelemetry.io/otel/semconv
```
and warnings about deprecated module
```
go: module github.com/golang/protobuf is deprecated: Use the "google.golang.org/protobuf" module instead.
```

### CODE REVIEW

File: benchmarks/online-shop/productcatalogservice/go.mod [hunk 1]
```diff
@@ -3,43 +3,37 @@ module github.com/GoogleCloudPlatform/microservices-demo/src/productcatalogservi
 go 1.18
 
 require (
```
Comment (ustiugov) [inline benchmarks/online-shop/productcatalogservice/go.mod:5]: @dhschall why do we have 2 `require(...)` statements?

File: benchmarks/online-shop/shippingservice/go.mod [hunk 2]
```diff
@@ -3,43 +3,37 @@ module github.com/GoogleCloudPlatform/microservices-demo/src/shippingservice
 go 1.18
 
 require (
```
Comment (ustiugov) [inline benchmarks/online-shop/shippingservice/go.mod:5]: same here

**Review by ustiugov — COMMENTED** [review]
LGTM. @dhschall could you take a look? also, could you run a couple of your workloads to see if they still work? the CI is currently in a really bad shape... need to fix (but not for this PR)

File: benchmarks/online-shop/productcatalogservice/go.mod (hunk 1 above)
Comment (sk1tter) [reply benchmarks/online-shop/productcatalogservice/go.mod:5]: Oh these are go 1.18. I think from go 1.17 it became that the first `require` statement would have direct dependencies and the second would only have transitive dependencies.

Comment (ustiugov) [reply benchmarks/online-shop/productcatalogservice/go.mod:5]: hmm... ok maybe we should consider going to 1.18 after all.... @dhschall what do you think?

Comment (dhschall) [reply benchmarks/online-shop/productcatalogservice/go.mod:5]: Yes, we should. vHive is 1.18
I already started to move some of my stuff as I run into a version problem when I wanted to install vHive

**Comment by ustiugov** [comment]
@dhschall can we merge this?

**Review by dhschall — APPROVED** [review]

**Pull request merged**